tsl/src/remote/dist_copy.c

```diff
@@ -15,31 +17,35 @@
 #include <utils/lsyscache.h>
 
 #include "compat/compat.h"
-#include "dist_copy.h"
+#include "chunk.h"
 #include "copy.h"
-#include "nodes/chunk_dispatch.h"
-#include "nodes/chunk_insert_state.h"
+#include "data_node.h"
 #include "dimension.h"
+#include "dimension_slice.h"
+#include "guc.h"
+#include "hypercube.h"
 #include "hypertable.h"
+#include "nodes/chunk_dispatch.h"
+#include "nodes/chunk_insert_state.h"
 #include "partitioning.h"
-#include "chunk.h"
-#include "ts_catalog/chunk_data_node.h"
-#include "guc.h"
 #include "remote/connection_cache.h"
 #include "remote/dist_txn.h"
-#include "data_node.h"
+#include "ts_catalog/chunk_data_node.h"
 
 #define DEFAULT_PG_DELIMITER '\t'
 #define DEFAULT_PG_NULL_VALUE "\\N"
 
-/* This will maintain a list of connections associated with a given chunk so we don't have to keep
- * looking them up every time.
+/*
+ * Maximum number of rows in batch for insert. Note that arrays of this size are
+ * also allocated on stack.
  */
-typedef struct ChunkConnectionList
-{
-	int32 chunk_id;
-	List *connections;
-} ChunkConnectionList;
+#define MAX_BATCH_ROWS 1024
+
+/*
+ * Maximum bytes of COPY data in batch. This is also the default size of the
+ * output copy data buffer.
+ */
+#define MAX_BATCH_BYTES 10 * 1024 * 1024
```

Comment:
IDK, postgres uses a define for this as well (MAX_BUFFERED_BYTES, copyfrom.c). I don't see an obvious need to make this configurable. Agree that it might be good in principle for advanced users, but on the other hands we shouldn't have too much options or the product becomes hard to use.